src/modules/ssh_docker/mappers/docker_container.ts

```diff
@@ -0,0 +1,231 @@
+import Modem from 'docker-modem';
+import Docker, { ContainerInspectInfo, DockerOptions } from 'dockerode';
+import _ from 'lodash';
+import SSH2Promise from 'ssh2-promise';
+
+import { Context, Crud, MapperBase } from '../../interfaces';
+import { sshAccounts } from '../../ssh_accounts';
+import { SshCredentials } from '../../ssh_accounts/entity';
+import { DockerContainer, dockerContainerStates } from '../entity';
+import { SshDocker } from '../index';
+
+export class DockerContainerMapper extends MapperBase<DockerContainer> {
+  module: SshDocker;
+  entity = DockerContainer;
+  equals = (a: DockerContainer, b: DockerContainer) => {
+    return (
+      Object.is(a.name, b.name) &&
+      Object.is(a.image, b.image) &&
+      _.isEqual(a.env, b.env) &&
+      _.isEqual(a.command, b.command) &&
+      _.isEqual(a.entrypoint, b.entrypoint) &&
+      Object.is(a.created?.getTime(), b.created?.getTime()) &&
+      _.isEqual(a.ports, b.ports) &&
+      _.isEqual(a.labels, b.labels) &&
+      Object.is(a.state, b.state) &&
+      _.isEqual(a.volumes, b.volumes) &&
+      _.isEqual(a.mounts, b.mounts) &&
+      _.isEqual(a.binds, b.binds)
+    );
+  };
+
+  private containerMapper(serverName: string, container: ContainerInspectInfo): DockerContainer {
+    const out: DockerContainer = new DockerContainer();
+    out.serverName = serverName;
+    out.containerId = container.Id;
+    out.name = container.Name.startsWith('/') ? container.Name.slice(1) : container.Name;
+    out.image = container.Config.Image;
+    out.env = container.Config.Env;
+    out.command = container.Config.Cmd;
+    if (container.Config.Entrypoint) {
+      if (!Array.isArray(container.Config.Entrypoint)) out.entrypoint = [container.Config.Entrypoint];
+      else out.entrypoint = container.Config.Entrypoint;
+    }
+    out.created = new Date(container.Created);
+    out.ports = container.HostConfig.PortBindings ?? undefined;
+    out.labels = container.Config.Labels;
+    out.state = container.State.Status;
+    out.volumes = Object.keys(container.Config.Volumes ?? {}).sort() ?? undefined;
```

Comment:
I suggested this, it's so the array can be compared like a set immediately and a difference is just the first time the two arrays differ in an `O(n)` check instead of an `O(n^2)` check